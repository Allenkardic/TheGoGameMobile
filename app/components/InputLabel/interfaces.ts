import React from 'react';

export type IInputLabelProps = React.FC<{
  children: string;
}>;
