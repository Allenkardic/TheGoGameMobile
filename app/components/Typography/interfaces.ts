import {TextProps as RNTextProps} from 'react-native';
import {HeaderTitleProps as NavHeaderTitleProps} from '@react-navigation/elements';

export type FontWeight = 'light' | 'medium' | 'bold' | 'semi-bold' | 'black';

export type FontColor =
  | 'alt'
  | 'primary'
  | 'muted'
  | 'primary'
  | 'gray'
  | 'info'
  | 'darkText'
  | 'lightText';

export type FontSize = 'xsmall' | 'small' | 'base' | 'large' | 'xlarge';

export interface TextProps extends RNTextProps {
  weight?: FontWeight;
  /**
   * @deprecated please switch to using weight="black"
   */
  stylish?: boolean;
  color?: FontColor;
  fontSize?: FontSize;
  garnettFont?: boolean;
}

export interface HeaderTitleProps extends NavHeaderTitleProps {
  fontSize?: 'default' | 'small';
}
