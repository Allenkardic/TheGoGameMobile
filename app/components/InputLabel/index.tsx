import React, {useMemo, memo} from 'react';
import {StyleSheet} from 'react-native';
import {Colors} from '../../utils';
import Text from '../Typography';
import {IInputLabelProps} from './interfaces';

const InputLabel: IInputLabelProps = function InputLabel({children}) {
  const style = useMemo(
    () => [styles.text, {color: Colors.inputLabelText}],
    [Colors.inputLabelText],
  );
  return <Text style={style}>{children}</Text>;
};

const styles = StyleSheet.create({
  text: {
    marginBottom: 8,
    fontSize: 14,
    lineHeight: 17,
    color: '#4E5158',
  },
});

export default memo(InputLabel);
