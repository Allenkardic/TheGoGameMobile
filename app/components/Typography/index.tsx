import React, {useMemo, memo} from 'react';
import {Text as RNText, StyleSheet} from 'react-native';
import {Colors} from '../../utils';
import {TextProps, FontWeight, FontSize} from './interfaces';

/**
 * This gets the font style for the text component,
 * defaults to book font style.
 * @param weight FontWeight
 */
const getInterFontWeight = (weight?: FontWeight) => {
  switch (weight) {
    case 'light':
      return styles.interLight;
    case 'medium':
      return styles.interMedium;
    case 'semi-bold':
      return styles.interSemiBold;
    case 'bold':
      return styles.interBold;
    case 'black':
      return styles.interBlackWeight;
    default:
      return null;
  }
};

/**
 * This gets the font style for the text component,
 * defaults to book font style.
 * @param weight FontWeight
 */

const getFontSize = (size?: FontSize) => {
  switch (size) {
    case 'xsmall':
      return styles.fontXsmall;
    case 'small':
      return styles.fontSmall;
    case 'base':
      return styles.fontBase;
    case 'large':
      return styles.fontLarge;
    case 'xlarge':
      return styles.fontXlarge;
    default:
      return null;
  }
};
export const Text: React.FC<TextProps> = function Text({
  weight,
  style,
  color,
  fontSize,
  ...props
}) {
  const textColor = useMemo(() => {
    switch (color) {
      case 'alt':
        return Colors.altText;
      case 'muted':
        return Colors.muted;
      case 'primary':
        return Colors.primary;
      case 'gray':
        return Colors.grey;
      case 'info':
        return Colors.infoText;
      case 'darkText':
        return Colors.darkText;
      case 'lightText':
        return Colors.lightText;
      default:
        return Colors.text;
    }
  }, [
    color,
    Colors.altText,
    Colors.grey,
    Colors.infoText,
    Colors.infoText,
    Colors.muted,
    Colors.primary,
    Colors.text,
    Colors.darkText,
    Colors.lightText,
  ]);

  const fontStyle = useMemo(() => {
    return getInterFontWeight(weight);
  }, [weight]);

  const size = useMemo(() => {
    return getFontSize(fontSize);
  }, [fontSize]);

  const combinedStyle = useMemo(() => {
    return [styles.base, {color: textColor}, style, fontStyle, size];
  }, [fontStyle, style, textColor, size]);

  return <RNText style={combinedStyle} {...props} testID="text" />;
};

const styles = StyleSheet.create({
  base: {
    fontSize: 16,
    lineHeight: 24,
    fontFamily: 'Inter-Regular',
  },
  small: {
    fontSize: 14,
    lineHeight: 21,
    fontFamily: 'Inter-Regular',
  },
  interLight: {
    fontFamily: 'Inter-Light',
  },
  interMedium: {
    fontFamily: 'Inter-Medium',
  },
  interSemiBold: {
    fontFamily: 'Inter-Bold',
  },
  interBold: {
    fontFamily: 'Inter-Bold',
  },
  interBlackWeight: {
    fontFamily: 'Inter-Black',
  },
  fontXsmall: {
    fontSize: 11,
    lineHeight: 19,
  },
  fontSmall: {
    fontSize: 14,
    lineHeight: 21,
  },
  fontBase: {
    fontSize: 16,
    lineHeight: 24,
  },
  fontLarge: {
    fontSize: 18,
    lineHeight: 24,
  },
  fontXlarge: {
    fontSize: 28,
    lineHeight: 27.72,
  },
});

export default memo(Text);
