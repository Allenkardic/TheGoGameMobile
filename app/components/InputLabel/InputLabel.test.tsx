import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from '../Typography';
import InputLabel from '.';

describe('InputLabel Component', () => {
  it('renders correctly with label text', () => {
    const component = renderer.create(<InputLabel>Test Label</InputLabel>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays the correct label text', () => {
    const component = renderer.create(<InputLabel>Test Label</InputLabel>);
    const root = component.root;
    expect(root.props.children).toBe('Test Label');
  });
});
