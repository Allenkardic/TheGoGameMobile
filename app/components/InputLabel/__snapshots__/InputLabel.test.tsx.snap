// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel Component renders correctly with label text 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Inter-Regular",
        "fontSize": 16,
        "lineHeight": 24,
      },
      {
        "color": "#010a1a",
      },
      [
        {
          "color": "#4E5158",
          "fontSize": 14,
          "lineHeight": 17,
          "marginBottom": 8,
        },
        {
          "color": "#1B1C20",
        },
      ],
      null,
      null,
    ]
  }
  testID="text"
>
  Test Label
</Text>
`;
